File: jscomp/test/lib_js_test.ml




external to_json_string : 'a -> string = "JSON.stringify"
[@@bs.val]


let () =
  begin

    Js.log @@ to_json_string [1;2;3];
    Js.log "hey";
  end
external of_any : 'a -> string = "String"
[@@bs.val]
let suites = Mt.[
    "anything_to_string", (fun _ -> Eq("3", of_any 3 ));
    (* in js, array is printed as {[ 1,2 ]} without brackets *)    
    (* "array_to_string", (fun _ -> Eq("[0]", Js.anything_to_string [|0|]))     *)
]
;; Mt.from_pair_suites __MODULE__ suites

File: jscomp/test/lib_js_test.js
'use strict';

var Mt = require("./mt.js");

console.log(JSON.stringify({
          hd: 1,
          tl: {
            hd: 2,
            tl: {
              hd: 3,
              tl: /* [] */0
            }
          }
        }));

console.log("hey");

var suites_0 = [
  "anything_to_string",
  (function (param) {
      return {
              TAG: "Eq",
              _0: "3",
              _1: String(3)
            };
    })
];

var suites = {
  hd: suites_0,
  tl: /* [] */0
};

Mt.from_pair_suites("Lib_js_test", suites);

exports.suites = suites;
/*  Not a pure module */
